### DISCUSSION TITLE
Introducing Scoliolyzer: A New Tool for Scoliosis Analysis Using Ultralytics YOLO

### DISCUSSION THREAD

**Discussion by Hamer1224** [ultralytics/ultralytics#22412] [Solutions] [answered]
Hi everyone,

I'm excited to share a project I've been working on called Scoliolyzer, a tool for analyzing X-ray images to measure the Cobb angle for scoliosis assessment. This project is built using the powerful and versatile Ultralytics YOLO models.

What is Scoliolyzer?

Scoliolyzer is a user-friendly application designed to assist medical professionals and researchers in the diagnosis and monitoring of scoliosis. By leveraging the object detection capabilities of YOLO, Scoliolyzer can:

Automatically detect and measure the Cobb angle from an X-ray image.

Provide a visual representation of the spine's curvature.

Streamline the process of scoliosis assessment, saving time and improving accuracy.

You can check out the project on GitHub:

 https://github.com/Hamer1224/Scoliolyzer

Why Ultralytics?

I chose to build Scoliolyzer with Ultralytics YOLO because of its:

High accuracy and performance in object detection.
Ease of use and implementation, which allowed me to focus on the core features of the app.
Active and supportive community, which has been a great resource throughout the development process.

We Need Your Feedback!

I would love to get your feedback on Scoliolyzer. Whether you're a developer, a medical professional, or just someone interested in the project, your insights and suggestions would be invaluable.
Please feel free to check out the repository, try out the app, and let me know what you think. I'm also open to contributions if you're interested in helping to improve the project.

Thanks for your time, and I look forward to hearing from you!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Hamer1224, thank you for sharing Scoliolyzer and for your interest in Ultralytics 🚀! This is an automated response to help you get the most from our resources while an Ultralytics engineer reviews and follows up soon.

Great to hear you’re leveraging Ultralytics YOLO for medical imaging. For general guidance, we recommend a visit to the [Docs](https://docs.ultralytics.com/) where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question or a showcase seeking feedback, please share details such as:
- Model variant used (e.g., YOLO11, YOLO26), training settings, and dataset info
- A few input/output examples or screenshots demonstrating Cobb angle detection
- Evaluation metrics and any specific challenges you’d like feedback on
- Environment details and logs if you encountered any issues

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thanks again for showcasing your project—excited to see more details and community feedback! 🙌

**Reply by glenn-jocher** [reply, marked as answer]
Thanks for sharing Scoliolyzer—great to see Ultralytics YOLO applied to Cobb angle measurement. For robust angles, consider training YOLO11 for OBB or Pose so you can derive vertebral endplate orientation directly; the workflow and examples are outlined in the [YOLO11 OBB guide](https://www.ultralytics.com/blog/how-to-use-ultralytics-yolo11-for-obb-object-detection) and the OBB section of our [computer vision tasks docs](https://docs.ultralytics.com/tasks/). To compute the Cobb angle from two endplate orientations (degrees), you can use:

```python
def cobb_angle(theta1_deg: float, theta2_deg: float) -> float:
    d = abs((theta1_deg - theta2_deg) % 180.0)
    return 180.0 - d if d > 90.0 else d
```

If you’d like targeted feedback, please post a few anonymized X-rays with predicted boxes/keypoints and your current angle outputs here in this thread for the community to review; we don’t offer private support.
